pkg/streamingpromql/testdata/ours/binary_operators.test

```diff
@@ -98,11 +98,21 @@ eval range from 0 to 24m step 6m left_side - on(env, pod) right_side
   {env="test", pod="a"} -9 -18 -27
   {env="test", pod="b"} -36 -45 -54
 
+eval range from 0 to 24m step 6m left_side - on(pod, env) right_side
```

Comment:
(nit) should add a comment here on the purpose of these tests and that the load order needs to be preserved

Reply:
>that the load order needs to be preserved

Not sure I follow this part - what are you referring to by "load order" here?

Reply:
the order the series are loaded in (above on line 88) could affect this test right? It's possible they are loaded in a way where the series are already sorted after the on/etc.

Reply:
It doesn't matter what order the series are in - the only thing that matters is the order of the label names passed to `BytesWithLabels` or `BytesWithoutLabels`.

(and regardless of the order of the series in the `load`, they'll be retrieved in sorted order due to how chunks streaming works)